gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -112,20 +90,25 @@ nns_edge_release_handle (nns_edge_h edge_h)
 
   eh = (nns_edge_handle_s *) edge_h;
 
-  if (!nns_edge_handle_is_valid (edge_h)) {
+  if (!NNS_EDGE_MAGIC_IS_VALID (eh)) {
     nns_edge_loge ("Invalid param, given edge handle is invalid.");
     return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
 
   if (eh->event_cb) {
     /** @todo send new event (release handle) */
-    eh->event_cb = NULL;
   }
 
+  nns_edge_lock (eh);
```

Comment:
Doesn't ```event_cb``` need protection?

Reply:
Possible synchronization error: you are protecting ```magic``` here while there are functions not protecting it with the lock.

Reply:
1. will do in next PR (release event)
2. offline discussion later